lesson06/task3/server/main.go

```diff
@@ -0,0 +1,63 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"time"
+)
+
+var name string
+var address string
+
+func handler(w http.ResponseWriter, r *http.Request) {
+
+	if r.URL.Path != "/" {
+		http.Error(w, "404 not found.", http.StatusNotFound)
+		return
+	}
+
+	switch r.Method {
+	case "GET":
+
+		// GET return HTML page
+		http.ServeFile(w, r, "form.html")
+
+	case "POST":
+
+		if err := r.ParseForm(); err != nil {
+			fmt.Fprintf(w, "ParseForm() err: %v", err)
+			return
+		}
+		name = r.FormValue("name")
+		address = r.FormValue("address")
+
+		expire := time.Now().Add(200 * time.Minute)
+		cookie := &http.Cookie{
+			Name:    "token",
+			Value:   fmt.Sprintf("%s:%s", name, address),
+			MaxAge:  300,
+			Expires: expire,
+		}
+
+		http.SetCookie(w, cookie)
+
+		w.WriteHeader(200)
+		w.Write([]byte(fmt.Sprintf("Cookie name: %s is setted successful\n", cookie.Name)))
```

Comment:
You need to remove this line as well as the line 49 so that the page is displayed correctly. Right now you add these two lines to the response, which makes an invalid html and the page is not rendered by the browser

Reply:
Magic start to work. Thanks a lot!